### PR TITLE
Lizard "Rework"

### PR DESCRIPTION
**Pull request by LankLTE**
## About the PR

Part of my Species Reworks, this gives lizards the ability to pull without hands, a diet restriction (Meat, Fruit, and Pills), and removes their random damage modifiers while toning up / down their notable ones to reasonable levels. Additionally normalizes their specific heat, removes their extra cold damage taken, and removes their random walking speed increase. 

## Why / Balance

To make lizards more unique as a species and remove their random damage modifiers, as discussed in [this](https://hackmd.io/-CnGYVtPQ2CHOJFp_ZYclA) design doc. 
This does outright disallow them from eating any other food (akin to moths). I'm not a *huge* fan of this implementation, but there is literally no way to do this otherwise that doesn't involve outright rewriting how food works. 

## Technical details

Any food I could find that was a fruit / meat has been tagged appropriately. This shouldn't break anything but it's possible (likely) I've missed a few, which can be fixed later. 

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Lank
- tweak: Reptilians have received a minor rework. Notable features include a diet restriction to only meat, fruit and pills, and the ability to pull objects using their tail.

### CODE REVIEW

**Comment by BolloTea** [comment]
Why didn’t you like the cold?

**Comment by LankLTE** [comment]
> Why didn’t you like the cold?

They still take cold damage sooner, they just don't take more. I don't know why I laid it out like that, I'd honestly be fine changing that back.

**Review by Emisse — APPROVED** [review]

**Comment by Emisse** [comment]
i approve the stat changes n shit idk if this is the sane way to do the food but for now fuck it w/e

**Comment by LankLTE** [comment]
Will fix test fails tomorrow

**Comment by BolloTea** [comment]
> > Why didn’t you like the cold?
> 
> They still take cold damage sooner, they just don't take more. I don't know why I laid it out like that, I'd honestly be fine changing that back.

Thanks for the clarification, otherwise it sounded a little strange)

**Comment by LankLTE** [comment]
> Will fix test fails tomorrow

I lied I'm doing it right now. 

> Thanks for the clarification, otherwise it sounded a little strange)

No problem. I also remember now that the point of it was to make emergency suits not completely unusable. You'll still die a lot faster, but not like, 3x as fast like you would before.

**Comment by UbaserB** [comment]
i dont like the fact lizards cant eat pills please thats the only thing i wish for to be changed, considering my most important and main character is a lizard (im begging you)

**Comment by LankLTE** [comment]
> i dont like the fact lizards cant eat pills please thats the only thing i wish for to be changed, considering my most important and main character is a lizard (im begging you)

I don't know if I made it clear enough but when I said "meat, fruit and pills" I mean that's *all* they can eat- so they can already eat pills still.

**Comment by VasilisThePikachu** [comment]
I still really don't like the diet restrictions here, at least outright disallowing it. Would be better for the future to do a rework of food at this point since this just kinda sounds like a temporary "fix" that will just stick around in the end.

**Comment by UbaserB** [comment]
> > i dont like the fact lizards cant eat pills please thats the only thing i wish for to be changed, considering my most important and main character is a lizard (im begging you)
> 
> I don't know if I made it clear enough but when I said "meat, fruit and pills" I mean that's _all_ they can eat- so they can already eat pills still.

oh ok lol

**Comment by IProduceWidgets** [comment]
Lizards are already poisoned by chocolate, would it make sense to extend that as a food limitation rather than simply stopping them from eating altogether?

**Comment by VasilisThePikachu** [comment]
> Lizards are already poisoned by chocolate, would it make sense to extend that as a food limitation rather than simply stopping them from eating altogether?

Thats kinda part of the "food rework" they were talking about. With the way food is coded right now its kinda hard. Plus making it toxic sounds kinda worse tbh. In 13 they can eat everything but stuff like grain or something like that gave you a mood debuff. We wont have moods here anytime soon... soooo

**Comment by LankLTE** [comment]
Also, I think poisoning them is sort of unintuitive for new players because, at least with how it’s currently implemented for things like chocolate, people would just go to eat something they could as any other race then suddenly start dying. That seems weird when we have the ability to stop them from eating them anyways.

**Comment by Hmeister-fake** [comment]
Maybe if the cold damage modifier was replaced by something more unique like being getting cold over time (cold blooded and all, don't generate heat) and being unable to gain heat without a space pen or other drugs, warm clothing, or maybe make it so that when they get hugged they get a little warmer

**Comment by JoeHammad1844** [comment]
why did you reduce their heat damage so much?

**Comment by LankLTE** [comment]
> why did you reduce their heat damage so much?

They used to have fairly good resistances which were removed, and this current resistances set mirrors moths (0.7 cold and 1.3 heat). I know heat is more relevant in combat, but given cold is the more common type environmentally I feel it'll balance out (and it's really easy to fix if it doesn't)

**Comment by JoeHammad1844** [comment]
that makes sense but 30% resistance still feels a like little too much, although you'll mostly only encounter it when getting shot with a laser, but a lizard wearing a sec vest will, if my math isnt wrong, add up to 60% resistance, which is just as good as an elite hardsuit and 90% with an elite hardsuit

**Comment by LankLTE** [comment]
> a lizard wearing a sec vest will, if my math isnt wrong, add up to 60% resistance, which is just as good as an elite hardsuit and 90% with an elite hardsuit

Your math is wrong. Not only does that not check out additively (sec vest is 20% heat resist), damage modifiers in this game are multiplicative, so a lizard with a sec vest comes out to about 45% resistance

**Comment by JoeHammad1844** [comment]
so 30% multiplied by 20%

**Comment by LankLTE** [comment]
> so 30% multiplied by 20%

The damage coefficients are multiplied, not the resistance percentages. 0.7 (species) * 0.8 (vest) to get 0.56 (44% resistance)

**Comment by LankLTE** [comment]
Tests work locally- fairly certain test fail is erroneous.

**Comment by LankLTE** [comment]
Don't mind that, I'm just an idiot.

**Review by crazybrain23 — CHANGES_REQUESTED** [review]
Nearly every time you added a tag, you've overridden the previously inherited tags by not including them, which is causing the test errors (among other bugs).

**Comment by LankLTE** [comment]
> Nearly every time you added a tag, you've overridden the previously inherited tags by not including them, which is causing the test errors (among other bugs).

I had just hoped there weren't any. Damnit.

**Comment by crazybrain23** [comment]
> > Nearly every time you added a tag, you've overridden the previously inherited tags by not including them, which is causing the test errors (among other bugs).
> 
> I had just hoped there weren't any. Damnit.

Pizza and bread are the two I checked, but there's probably more.

**Comment by LankLTE** [comment]
holt shit it works

**Comment by crazybrain23** [comment]
> Pizza and bread are the two I checked, but there's probably more.

Pie and soup also need fixing.

**Comment by LankLTE** [comment]
> Pie and soup also need fixing.

fixed 👍

**Comment by LankLTE** [comment]
Think the tags are all fixed / it's good

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml [hunk 1]
```diff
@@ -980,6 +980,8 @@
     - Trash
     - VimPilot
     - Mouse
+    - DoorBumpOpener
```
Comment (Partmedia) [inline Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:983]: Why was this added here?

Comment (LankLTE) [reply Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:983]: Probably an accident.

Comment (LankLTE) [reply Resources/Prototypes/Entities/Mobs/NPCs/animals.yml]: Have removed it now, sorry

**Review by Partmedia — APPROVED** [review]

**Pull request merged**